src/openforms/registrations/contrib/objects_api/migrations/0011_create_objecttypesypes_service_from_url.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 3.2.23 on 2024-01-31 14:20
+import logging
+import re
+from furl import furl
+
+from django.db import migrations
+from django.db.migrations.state import StateApps
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+
+from openforms.registrations.contrib.objects_api.plugin import (
+    PLUGIN_IDENTIFIER as OBJECTS_API_PLUGIN_IDENTIFIER,
+)
+
+from zgw_consumers.constants import APITypes
+
+logger = logging.getLogger(__name__)
+
+
+def create_objecttypes_api_service(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    """Create a "placeholder" Objecttypes API service from an already existing Objecttypes URL.
+
+    This migration exists to help users migrate to a new version of Open Forms. Note that
+    they will still need to configure authentication on the created Objecttypes Service.
+    """
+    ObjectsAPIConfig = apps.get_model("registrations_objects_api", "ObjectsAPIConfig")
+
+    # First, try getting the default objecttypes URL
+    try:
+        objects_api_config = ObjectsAPIConfig.objects.get()
+    except ObjectsAPIConfig.DoesNotExist:
+        return
+    if objects_api_config.objecttypes_service:
+        # Extra safety, even though it shouldn't happen
+        return
+
+    objecttypes_url = objects_api_config.objecttype
+
+    # If no default objecttypes URL was configured, get it from form registration backends:
+    if not objecttypes_url:
+        form_registration_bases_urls: set[str] = set()
+
+        FormRegistrationBackend = apps.get_model("forms", "FormRegistrationBackend")
+        for form_registration_backend in FormRegistrationBackend.objects.filter(
+            backend=OBJECTS_API_PLUGIN_IDENTIFIER
+        ).iterator():
+            if url := form_registration_backend.options.get("objecttype"):
+                index = url.find("/api/")
+                if index != -1:
+                    objecttypes_url = url
+                    form_registration_bases_urls.add(url[:index])
+
+        if len(form_registration_bases_urls) >= 2:
+            logger.warning(
+                "Found %d different Objecttypes URLs in form registration backend options",
+                len(form_registration_bases_urls),
+            )
+
+    if objecttypes_url:
+        Service = apps.get_model("zgw_consumers", "Service")
+
+        try:
+            base, _ = re.split(r"api/v\d", objecttypes_url, 1)
+        except ValueError:
+            logger.warning(
+                "URL %s is not compliant, skipping migration", objecttypes_url
+            )
+            return
+
+        api_version = re.search(r"/api/v(\d)", objecttypes_url).group(1)
+        base += f"api/v{api_version}/"
```

Comment:
I still think this is overly complicated and doesn't require regex since we only need to account for the current situation of available versions (as the mgiration only applies to existing instances), but sure, let's go with this.

Also, note that `\d` in python regex also captures non-arabic numerals and can lead to interesting surprises :) 

```py
>>> malayalam = "൦"
>>> import re
>>> re.search(r"\d", malayalam)
<re.Match object; span=(0, 1), match='൦'>
```

Reply:
Yes but I wasn't sure if the v1 API was still used or not. If both v1 and v2 can exist, how should this be handled then? Should the endpoint be forced to use v2?